src/App.tsx

```diff
@@ -1,223 +1,442 @@
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
-
-            <form>
-              <input
-                data-cy="TodoTitleField"
-                type="text"
-                className="todo__title-field"
-                placeholder="Empty todo will be deleted"
-                defaultValue="JS"
-              />
-            </form>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">React</span>
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">Redux</span>
+          {todosAvailable
+          && (
+            <TodoList
+              todos={filteredTodos}
+              setTodos={setTodos}
+              editTodoStatus={updateTodoStatus}
+              selectedId={selectedId}
+              setSelectedId={setSelectedId}
+              title={title}
+              setTitle={setTitle}
+              editTodoTitle={editTodoTitle}
+              deleteTodo={deleteTodo}
+            />
+          )}
+        </section>
+        {todosAvailable && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="todosCounter">
+              {`${
+                todos.length
+                - todos.filter(todo => todo.completed).length} items left`}
+            </span>
+            <nav className="filter" data-cy="Filter">
+              <a
+                data-cy="FilterLinkAll"
```

Comment:
Will be good practice to create separate component for each section (header, section, footer, etc..)